src/Database/PostgreSQL/Simple/Transaction.hs

```diff
@@ -166,8 +166,7 @@ withTransactionModeRetry mode shouldRetry conn act =
         beginMode mode conn
         r <- act'
         case r of
-            Left e -> do
-                rollback_ conn
+            Left e ->
```

Comment:
as you refactored this way, which looks good; is it enough to catch `SqlError`, to avoid explicit `E.fromException`.

Small style improvement.

Reply:
Good idea.

Reply:
Changed.